File: 0001-two-sum/solution.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        nums_map = {}

        for i in range(len(nums)):
            t = target - nums[i]
            if t in nums_map:
                return [nums_map[t], i]
            nums_map[nums[i]] = i

        return []

File: 0001-two-sum/solution.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer, Integer> m = new HashMap<>();
        int[] res = new int[2];
        int n = nums.length;

        for (int i = 0; i < n; i++) {
            int t = target - nums[i];
            if (m.containsKey(t)) {
                res[0] = i;
                res[1] = m.get(t);
                return res;
            }
            m.put(nums[i], i);
        }

        return res;
    }
}
